mysqld_exporter.go

```diff
@@ -52,6 +63,56 @@ func NewMySQLExporter(dsn string) *Exporter {
 			Help:      "The last scrape error status.",
 		}),
 		metrics: map[string]prometheus.Gauge{},
+		commands: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "commands_total",
+			Help:      "Number of executed mysql commands.",
+		}, []string{"command"}),
+		connection_errors: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "connection_errors_total",
+			Help:      "Number of mysql connection errors.",
+		}, []string{"error"}),
+		handlers: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "handlers_total",
+			Help:      "Number of executed mysql handlers.",
+		}, []string{"handler"}),
+		innodb_buffer_pool: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "innodb_buffer_pool_total",
+			Help:      "Mysql Innodb buffer pool metrics.",
+		}, []string{"metric"}),
+		innodb_data: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "innodb_data_total",
+			Help:      "Mysql Innodb data operations.",
+		}, []string{"operation"}),
+		innodb_pages: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "innodb_pages_total",
+			Help:      "Mysql Innodb page operations.",
+		}, []string{"operation"}),
+		innodb_rows: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "innodb_rows_total",
+			Help:      "Mysql Innodb row operations.",
+		}, []string{"operation"}),
+		performance_schema: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "performance_schema_total",
+			Help:      "Mysql instrumentations that could not be loaded or created due to memory constraints",
+		}, []string{"instrumentation"}),
+		ssl: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "ssl_total",
+			Help:      "Mysql SSL metrics.",
+		}, []string{"metric"}),
+		threads: prometheus.NewUntypedVec(prometheus.UntypedOpts{
+			Namespace: namespace,
+			Name:      "threads_total",
```

Comment:
Hmm, it actually also seems like this is a mix of counters and gauges, so not too suitable for a single metric:

```
| Threads_cached                                | 95               |
| Threads_connected                             | 149              |
| Threads_created                               | 1898             |
| Threads_running                               | 3                |
```